### ISSUE TITLE
Add an option to limit the size of the database?

### ISSUE THREAD

**Issue by prismz** [PrivateBin/PrivateBin#951]
Forgive me if this is already implemented. I know you can configure the size of each paste, but how would I limit the size of the database itself? My server isn't super huge, and I'd like to prevent my PrivateBin instance from ever growing too large. I know there's probably an option to do this for MySQL and other databases, but I'm looking for something for the filesystem model.

Perhaps something like [this](https://askubuntu.com/a/841306)?

**Comment by elrido** [comment]
The above link concerns itself with filesystem quotas, a standard feature of many Linux filesystems. This will work with PrivateBin, if you have root access to your hosting environment.

If you are using one of the database abstractions, you need to look up your database documentation. MySQL, for example has only very limited quota options on a per-database-engine level ([one setting for MyISAM and one for InnoDB](https://www.thegeekdiary.com/how-to-set-space-limits-for-mysql-for-database-schema-table/)).

The error when the instance is full (no disk space left, quota error or error inserting paste into database) won't necessarily be very descriptive to your users, tough. It would display the generic server error message upon creating a new paste:
https://github.com/PrivateBin/PrivateBin/blob/f9d78d62a00ce3985b817b129c8e86d185d85275/js/privatebin.js#L4746

To give you some idea on expected disk usage, here are some numbers from two instances I host, both have the default limit of 10 MiB per paste, the larger one doesn't support file upload, the smaller one has file uploads enabled:
- 1.2 GiB disk space used / 40272 pastes = 31.25 KiB average paste size
- 45 MiB disk space used / 539 pastes = 85.5 KiB average paste size

In other words, you can most likely drastically reduce the per paste size limit, for example down to 256 KiB, and reduce the risk of your disk running full too quickly. Even with file uploads enabled, the average office document or web sized image will still fit into that limit.

**Comment by rugk** [comment]
I'd also advice for just using the existing options to limit attachments or the paste size to limit the general size.
If you are concerned about DDOS attacks, you [can also adjust the traffic limit](https://github.com/PrivateBin/PrivateBin/wiki/Configuration#trafficlimit).

Also consider monitoring your disk size. That (professional, i.e. automated monitoring with tools used for it and warnings pushed to you in case of problems) is generally a good idea of course. It is probably better to monitor the size and how it is going to be used rather than adding a hard limit and not using all the resources of the server you have.

Given all these options you have, I don't deem a special options to be necessary in PrivateBin.
So as it seems this more of a support question has been answered, I thus close this issue. If anyone has further questions on that topic, feel free to comment here, again.
